mifosng-android/src/main/java/com/mifos/mifosxdroid/online/clientdetails/ClientDetailsFragment.java

```diff
@@ -600,15 +604,21 @@ public void showUploadImageProgressbar(boolean b) {
         }
     }
 
+    public void urlBuilder() {
+        url = new StringBuilder();
+        url.append(PrefManager.getInstanceUrl()
+                + "clients/"
+                + "clientId"
+                + "/images?maxHeight=120&maxWidth=120");
+        String tempUrlCopy = url.toString();
+        tokenPositionStart = tempUrlCopy.indexOf("clientId");
+        tokenPositionEnd = tokenPositionStart + 8;
```

Comment:
Good attempt abstracting out this method. 

You can do this simply like this

```
sb = StringBuilder(PrefManager.getInstanceUrl());
sb.append(String.format("clients/%d/images?maxHeight=120&maxWidth=120",clientId);
String url = sb.toString();
```

You don't need to do tokenPositionStart,End calculations and then a replace below.